corehq/apps/commtrack/models.py

```diff
@@ -503,20 +504,33 @@ def _yield_txns(section_id, quantity):
                     yield _txn(action=const.StockActions.RECEIPTS, case_id=dst,
                                section_id=section_id, quantity=quantity)
 
+        def _log_blank_quantity_error(config, section_id):
+            logging.error((
+                "Blank transaction quantity submitted on domain %s for "
+                "a %s ledger" % (config.domain, section_id)
+            ))
+
         section_id = action_node.attrib.get('section-id', None)
         grouped_entries = section_id is not None
         if grouped_entries:
-            quantity = float(product_node.attrib.get('quantity'))
-            for txn in _yield_txns(section_id, quantity):
-                yield txn
-
+            quantity = product_node.attrib.get('quantity')
+            # make sure quantity is not an empty, unset node value
+            if str(quantity).strip() != '':
+                for txn in _yield_txns(section_id, float(quantity)):
+                    yield txn
+            else:
+                _log_blank_quantity_error(config, section_id)
         else:
             values = [child for child in product_node]
             for value in values:
                 section_id = value.attrib.get('section-id')
-                quantity = float(value.attrib.get('quantity'))
-                for txn in _yield_txns(section_id, quantity):
-                    yield txn
+                quantity = value.attrib.get('quantity')
+                # make sure quantity is not an empty, unset node value
+                if str(quantity).strip() != '':
+                    for txn in _yield_txns(section_id, flaot(quantity)):
```

Comment:
.... Wow. I thought both paths were hit by the test, will add better coverage lol